Automattic/wp-calypso
# Store|Woo: Hide deprecated Store menu item for sites with Business plan subscriptions after Store was deprecated

Part of the "Sunset Store on WordPress.com" epic (https://github.com/woocommerce/woocommerce-admin/issues/5816) -- see that for more context.

#### Changes proposed in this Pull Request

* Updates to `canCurrentUserUseCalypsoStore` (was `canCurrentUserUseStore`) and `canCurrentUserUseWooCommerceCoreStore` (new) selectors
* Updates My Sites sidebar to hide "Store" menu item for sites with Business plan subscriptions started after Store was deprecated

#### Testing instructions

##### Automated testing

- Check out branch
- Verify that all automated tests for selectors pass:

```
yarn run test-client client/state/sites/test/selectors.js
yarn run test-client client/my-sites/sidebar/test/sidebar.js
```

##### Manual testing

_Note: to change the deprecation start date when testing, in `client/lib/plans/constants.js`, edit the `STORE_DEPRECATION_START_DATE` constant. (I don't like giving manual testing instructions that involve editing code, but this is the easiest way. There might also be a way to change the subscription date in the database/state tree.)_

- Check out branch

###### Verify that "Store" and "WooCommerce" menu items appear correctly for sites when Store is deprecated:

```
ENABLE_FEATURES=woocommerce/store-deprecated yarn start
```

- "Store" available for sites with Business plan subscriptions started before Store was deprecated
- "Store" not available for sites with Business plan subscriptions started after Store was deprecated
- "Store" (linking to WooCommerce Core) available for sites with eCommerce plan subscriptions
- "WooCommerce" available for sites with Business plan subscriptions

###### Verify that "Store" and "WooCommerce" menu items appear correctly for sites when Store is removed:

```
ENABLE_FEATURES=woocommerce/store-removed yarn start
```

- "Store" not available for sites with Business plan subscriptions
- "Store" (linking to WooCommerce Core) available for sites with eCommerce plan subscriptions
- "WooCommerce" available for sites with Business plan subscriptions

Fixes #48424
